torch/quantization/adaround_fake_quantize.py

```diff
@@ -0,0 +1,102 @@
+import torch
+import torch.nn.init as init
+import math
+from torch.quantization.fake_quantize import FakeQuantize
+from torch.quantization.observer import MovingAverageMinMaxObserver
+from torch.quantization.qconfig import *
+from torch.quantization.fake_quantize import *
+
+
+class AdaRoundFakeQuantize(FakeQuantize):
+    ''' This class is an extension of the FakeQuantize module and is used similarly.
+    The difference between the two is that you can use the continous_V attribute to alter the
+    default rounding scheme during quantization. The defualt is to round to the nearest number,
+    but applying an optimizer on this attribute can yield a better rounding scheme and improve
+    quantization results.
+
+    '''
+    adaround_default_attribute_settings = {'continous_V': None,
+                                           'beta_high': 8,
+                                           'beta_low': 2,
+                                           'norm_scaling': 10,
+                                           'regularization_scaling': .1}
+
+    def __init__(self, *args, **keywords):
+        settings = AdaRoundFakeQuantize.adaround_default_attribute_settings
+
+        for attribute in settings:
+            if attribute in settings:
+                setattr(self, attribute, keywords[attribute])
+                del keywords[attribute]
```

Comment:
why do we need to delete here?

Reply:
keywords gets passed to its parent class FakeQuantize, i thought weird things would happen if im passing extra parameters to FakeQuantize

Reply:
you should call super first with explicit arguments and then initialize the remaining attributes